### DISCUSSION TITLE
Adding a text node to the AST

### DISCUSSION THREAD

**Discussion by matthewmueller** [yuin/goldmark#215] [Q&A] [answered]
Hey there, 

I'd like to add a new section to the AST tree before rendering it. Something like the following:

```go
h1 := ast.NewHeading(1)
txt := ast.NewText()
txt.Text(value) // This is what I can't figure out.
h1.AppendChild(txt)
doc.AppendChild(h1)
```

Any ideas how I can add text? Thanks!

**Comment by yuin** [comment, marked as answer]
Use `ast.String` instead of `ast.Text`
